src/main/java/reposense/report/ErrorSummary.java

```diff
@@ -0,0 +1,38 @@
+package reposense.report;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Holds the data of list of repos that failed to analyze and the reasons for the failed operation.
+ */
+public class ErrorSummary {
+    private static ErrorSummary instance = null;
+    private static List<Map<String, String>> errorList = new ArrayList<>();
+
+    public static ErrorSummary getInstance() {
+        if (instance == null) {
+            instance = new ErrorSummary();
+        }
+        return instance;
+    }
```

Comment:
Why Singleton?

It's only used in `ReportGenerator` right?

Reply:
I was thinking that this feature could be extended to report other issues, such as invalid file name that couldn't be analyzed, etc